spec/features/work_packages/share_spec.rb

```diff
@@ -224,6 +227,50 @@ def inherited_member_roles(group:)
           .not_to include(not_shared_yet_with_group, richard)
       end
 
+      aggregate_failures "Inviting multiple users or groups at once" do
+        # Inviting multiple users at once
+        share_modal.invite_user([richard, another_not_shared_yet_with_user], 'Edit')
+
+        share_modal.expect_shared_count_of(9)
+
+        share_modal.expect_shared_with(another_not_shared_yet_with_user, 'Edit', position: 1)
+        share_modal.expect_shared_with(richard, 'Edit', position: 2)
+
+        # They can be removed again
+        share_modal.remove_user(richard)
+        share_modal.remove_user(another_not_shared_yet_with_user)
+
+        share_modal.expect_shared_count_of(7)
+
+        # Groups can be added simultaneously as well
+        share_modal.invite_user([not_shared_yet_with_group, empty_group], 'Comment')
+
+        share_modal.expect_shared_count_of(9)
+
+        share_modal.expect_shared_with(not_shared_yet_with_group, 'Comment', position: 1)
+        share_modal.expect_shared_with(empty_group, 'Comment', position: 2)
```

Comment:
This is where the other set of flickering specs surge from and I have some implementation suggestions for the feature:

Due to the way turbo stream actions are processed, they're an instruction set parsed from top to bottom. So if in the autocompleter we select (in order) user A, then user B. The shares get created in the controller in that same order: A, then B. When generating the turbo stream responses this is the same order they come in: prepend A, **then** prepend B. Meaning that we will be prepending users in an inverted order to the one we selected.

I feel it would be nice if the prepending reflects the order in which I added the users to the autocompleter from a UX perspective. If we do so, the tests are also just a tad easier to parse through because we can call precisely what you wrote:
```ruby
        # Selected in this order
        share_modal.invite_user([not_shared_yet_with_group, empty_group], 'Comment')

        share_modal.expect_shared_count_of(9)

        # And get prepended in the exact same order
        share_modal.expect_shared_with(not_shared_yet_with_group, 'Comment', position: 1)
        share_modal.expect_shared_with(empty_group, 'Comment', position: 2)
```

This would require a change to the other multiple invite expectations to match this behavior but would be a lot easier to read then in my opinion.

You can accomplish this behavioral change in the controller quite simply by setting:

```ruby
@shares = overall_result.map(&:result).reverse
```

Reply:
I agree. The sorting order is much more intuitive, if it matches the order in the autocompleter.